### PR TITLE
Feature/catalog controller refactor

### PR DESCRIPTION
**Pull request by blackrat**
Refactor to move Solr IndexTypes to a more sensible place and extend it to support CatalogConfig too.
Change CatalogConfig to allow simple list passing of default options and to always add an internationalised label to options.

### CODE REVIEW

**Comment by dannyberzon** [comment]
Is there any way we can rebase off develop and resolve the conflict?

File: willow/app/controllers/blacklight/concerns/commands.rb [hunk 1]
```diff
@@ -0,0 +1,92 @@
+module Blacklight
+  module Concerns
+    module Commands
+      extend ActiveSupport::Concern
+      include ::Solr::Concerns::IndexTypes
+
+      class_methods do
+        private
+        def decode_name_and_options(name, options={})
+          actual_name=name.keys.first
```
Comment (tripledub) [inline willow/app/controllers/blacklight/concerns/commands.rb]: Nit pick but put spaces around `=`

Comment (blackrat) [reply willow/app/controllers/blacklight/concerns/commands.rb]: Yeah. That's a habit I have. I don't tend to use unnecessary whitespace.

Comment (tripledub) [reply willow/app/controllers/blacklight/concerns/commands.rb]: Do you have `RuboCop` installed in `Rubymine`?

**Pull request merged**